### PR TITLE
Added color overlay

### PR DESCRIPTION
**Pull request by emilwestergren**
Added functionality for drawing colored overlays (lines, points, boxes) on YUV images.

### CODE REVIEW

**Comment by davidhesselbom** [comment]
Please summarize the purpose of this PR. What's a `color overlay`?

**Comment by emilwestergren** [comment]
Updated.

File: source/draw/PaintEngine.ooc [hunk 1]
```diff
@@ -22,13 +22,14 @@ import Color
...
 	color: ColorBgra { get set }
+	width: Float { get set }
 	alpha ::= this color alpha
-	alphaAsFloat ::= this alpha as Float / 255.0f
-	init: func@ (=color)
-	init: func@ ~withBgr (colorBgr: ColorBgr) {
-		this color = colorBgr toBgra()
-	}
-	setApha: func@ (alpha: UInt8) {
+	alphaAsFloat ::= this alpha as Float / 255.0
```
Comment (davidhesselbom) [inline source/draw/PaintEngine.ooc]: What happened to the `f`?

File: source/draw/PaintEngine.ooc [hunk 2]
```diff
@@ -22,13 +22,14 @@ import Color
...
+	init: func@ (=color, =width)
+	init: func@ ~color (color: ColorBgra) { this init(color, 1.0f) }
```
Comment (davidhesselbom) [inline source/draw/PaintEngine.ooc]: Or 
```ooc
init: func@ ~color (=color, width := 1.0f ) { this width = width }
```

File: source/draw/PaintEngine.ooc (hunk 1 above)
Comment (emilwestergren) [inline source/draw/PaintEngine.ooc]: No clue, does it matter?

**Comment by emilwestergren** [comment]
Updated.

File: source/draw/gpu/opengl/OpenGLContext.ooc [hunk 3]
```diff
@@ -67,32 +67,12 @@ OpenGLContext: class extends GpuContext {
...
-		this _linesShader color = FloatPoint3D new(0.0f, 0.0f, 0.0f)
 		this _linesShader projection = projection
+		this _linesShader color = pen color normalized
 		this _linesShader use()
-		this _renderer drawLines(positions, pointList count, 2, 3.5f)
-		this _linesShader color = FloatPoint3D new(1.0f, 1.0f, 1.0f)
-		this _linesShader use()
-		this _renderer drawLines(positions, pointList count, 2, 1.5f)
-	}
-	drawBox: func (box: FloatBox2D, projection: FloatTransform3D) {
```
Comment (davidhesselbom) [inline source/draw/gpu/opengl/OpenGLContext.ooc]: `drawBox` isn't needed anymore?

Comment (emilwestergren) [inline source/draw/gpu/opengl/OpenGLContext.ooc]: The layer above this uses drawLines instead when drawing boxes, so no.

File: source/draw/gpu/GpuCanvas.ooc [hunk 4]
```diff
@@ -40,21 +40,41 @@ GpuCanvasYuv420Semiplanar: class extends GpuSurface {
...
+			yuv := value color toYuv()
+			this _target y canvas pen = Pen new(ColorBgra new(yuv y, 0, 0, 255), this pen width)
+			this _target uv canvas pen = Pen new(ColorBgra new(yuv u, yuv v, 0, 255), this pen width)
+		}
+	}
+
 	init: func (=_target, context: GpuContext) {
 		super(this _target size, context, context defaultMap, IntTransform2D identity)
-		this _target uv canvas clearColor = ColorBgra new(128, 128, 128, 128)
+		this _target uv canvas pen = Pen new(ColorBgra new(128, 128, 128, 128))
```
Comment (davidhesselbom) [inline source/draw/gpu/GpuCanvas.ooc]: I know it was already like this, but, won't this set the canvas to 50% transparency?

**Comment by davidhesselbom** [comment]
@sebastianbaginski, can you take a look at this?

**Comment by sebastianbaginski** [comment]
It looks ok, nice to have a width as another property of `Pen`. I don't think I can spot anything more than you already did.

File: source/draw/gpu/GpuCanvas.ooc (hunk 4 above)
Comment (emilwestergren) [inline source/draw/gpu/GpuCanvas.ooc]: Only the two first channels in an UV-image will be affected by clear so no. I'm not sure if BGRA is converted to RGBA so all channels might as well be 128 to be safe.

**Pull request merged**